server/src/auth/auth.module.ts

```diff
@@ -0,0 +1,22 @@
+import { Module } from "@nestjs/common";
+import { MongooseModule } from "@nestjs/mongoose";
+import { User, UserSchema } from "./schemas/user.schema";
+import { AuthService } from "./auth.service";
+import { AuthController } from "./auth.controller";
+import { JwtModule } from "@nestjs/jwt";
+import { PassportModule } from "@nestjs/passport";
+import { JwtStrategy } from "./jwt.strategy";
+
+@Module({
+  imports: [
+    MongooseModule.forFeature([{ name: User.name, schema: UserSchema }]),
+    PassportModule,
+    JwtModule.register({
+      secret: process.env.JWT_SECRET,
+      signOptions: { expiresIn: "1h" },
+    }),
+  ],
+  providers: [AuthService, JwtStrategy],
```

Comment:
```suggestion
    JwtModule.registerAsync({
      global: true,
      imports: [ConfigModule],
      inject: [ConfigService],
      useFactory: (config: ConfigService) => ({
        secret: config.get<string>("JWT_SECRET"),
        signOptions: { expiresIn: "1h" },
      }),
    }),
  ],
  exports: [AuthService, JwtModule],
  providers: [AuthService, JwtStrategy],
```
테스트코드는 되는데, post man으로 확인해봤을때 회원가입은 동작하는데 로그인에서 `secretOrPrivateKey must have a value`에러가 뜨더라구요! 찾아보니 env 불러오는게 문제였는데 [해당 링크](https://velog.io/@daep93/Nestjs-secretOrPrivateKey-must-have-a-value)를 참고했습니다! 이제 로그인하면 토큰 잘 받아와집니당!!

Reply:
헉...제가 수정해야 했던 부분을 찾아주셨군요...서윤님 덕분에 바로 해결했습니다!
이 부분에 대해서는 제가 별도로 개발 위키에 적어볼게요!